cdk/lib/clamav-scanner-stack.ts

```diff
@@ -0,0 +1,75 @@
+import { Fn, Stack,
+         aws_ecr as ecr,
+         aws_ecs as ecs,
+         aws_iam as iam,
+         aws_ssm as ssm,
+         aws_s3 as s3,
+         aws_s3_notifications as s3n,
+         aws_lambda_nodejs as lambdaNodejs
+       } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { ClamavScannerStackProps } from './clamav-scanner-stack-props';
+import { parseEcrAccountId, parseEcrRegion } from './common-stack-funcs';
+import { ClamavScan } from './clamav-scan';
+
+export class ClamavScannerStack extends Stack {
+
+  readonly lambda: lambdaNodejs.NodejsFunction;
+
+  constructor(scope: Construct, id: string, props: ClamavScannerStackProps) {
+    super(scope, id, props);
+
+    const clamavRepo = ecr.Repository.fromRepositoryArn(this, 'clamavRepo', `arn:aws:ecr:${parseEcrRegion(props.envProps.REGISTRY)}:${parseEcrAccountId(props.envProps.REGISTRY)}:repository/${props.envProps.REPOSITORY}/clamav`);
+
+    const clamavTaskDef = new ecs.FargateTaskDefinition(this, 'clamavTaskDef', {
+      cpu: props.clamavTaskDef.taskCpu,
+      memoryLimitMiB: props.clamavTaskDef.taskMem,
+    });
+
+    // Allow clamavScan to manage S3 files
+    const pCkanCloudstorageContainerName = ssm.StringParameter.fromStringParameterAttributes(this, 'pCkanCloudstorageContainerName', {
+      parameterName: `/${props.environment}/opendata/ckan/cloudstorage_container_name`,
+    });
+    const bucketArn = `arn:aws:s3:::${pCkanCloudstorageContainerName.stringValue}`;
+
+    const clamavScanPolicyAllowCloudstorage = new iam.PolicyStatement({
+      actions: [
+                  "s3:DeleteObject",
+                  "s3:DeleteObjectTagging",
+                  "s3:GetObject",
+                  "s3:GetObjectTagging",
+                  "s3:HeadBucket",
+                  "s3:ListObjects",
+                  "s3:PutObject",
+                  "s3:PutObjectTagging",
+                ],
+      resources: [
+        bucketArn,
+        `${bucketArn}/*`,
+      ],
+      effect: iam.Effect.ALLOW,
+    });
```

Comment:
This can be done with grants instead https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html#grantwbrreadwbrwriteidentity-objectskeypattern